### PR TITLE
Balance: Rebalance of ship mass and handling across the board again

### LINKED ISSUE #5271: There is a lack of progression in speed related to ship sizes
**Is your feature request related to a problem? Please describe.**
Currently in game there is a progression as you increase the size of your ship. Generally, the bigger the ship within a tier, the more HP it has, the more DPS it can output, and the more outfit/weapon/engine space it has. One would also think though that the bigger your ship, the slower it goes. While it is roughly the case that larger ships are slower than smaller ships, it is very easy to equip a larger ship with engines that make it as fast as or faster than a much smaller ship. Therefore, there is currently no real progression in speed related to ship size. Increasing the size of your ship yields no downsides, as the one expected downside of lower speed does not exist.

**Related Issue Links**
Other issues that either speak directly to or are related to the balance between large and small ships:
#1932 was closed because turrets were given turn rates. While this helped to give smaller ships a chance, the issue of speed still exists.
#4034 The issue with smaller ships vs larger ships is related to the issue of fighters being underwhelming at the moment.
#4407 suggested using multipliers to make generators that are more attractive to smaller ships, but I suggested that we instead simply make smaller ships regenerate faster. (This is also something mentioned in #1932.)
#4697 discusses ways to make smaller ships have more of a chance against larger ships. 

**Describe the solution you'd like**
Rebalance the mass and drag of ships to have a much more drastic jump between categories of ships.

For example, right now the difference in the average mass of a human interceptor versus a human light warship is only +80%. From light to medium warship it's +87%, and from medium to heavy warship it's +126%.
In terms of drag, the progression is +127% from interceptor to light warship, +86% from light warship to medium warship, and +57% from medium warship to heavy warship.

By increasing the jump in mass and drag as you go up a ship category, we'd make it so that larger ships handle noticeably slower than smaller ships, being the trade off for more HP and DPS.

**Describe alternatives you've considered**
Another point of concern might be the progression in power between various types of engines. A ship with atomic engines will be going MUCH faster than a ship with ion engines of the same size. Perhaps we want to reduce the difference between the slowest and fastest engines. Currently, a tiny atomic thruster has nearly double the thrust per ton as a tiny ion thruster. (87% more to be exact.) This difference in power continues all the way up the size progression of engines. (The huge atomic thruster has 81% more thrust per ton than the huge ion thruster.)

And in terms of progression of power as you increase in engine size, at the same time that the average human heavy warship has 565% more drag and 661% more mass than a human interceptor, huge thrusters have about 1260% more thrust in the case of ion and plasma thrusters, and 807% more thrust in the case of atomic engines when compared to the tiny thruster of the same engine category. 

This is to point out that on average, **the power of human engines as you increase in size currently outpaces the increase in drag and mass.** Therefore, we either need to reduce the power of engines with size or increase the drag and mass more with size.

**Additional context**
Should we go the route of increasing the mass of ships, we need to remember that hit force and optical tracking are currently balanced assuming the average mass of ships of certain categories.

### PR DESCRIPTION
**Pull request by Quantumshark**
**Balance: resolves #5271**

## Summary
All engines' thrust and turn have been multiplied by 1.5 (to provide room for small ships to be made faster without their hull masses going negative).
Ships have received multiplicative increases to their net mass (achieved by tweaking hull mass, but taking into consideration that and outfit space), and equally to their drag (top speed to acceleration ratios remain unchanged here, and could be tweaked as desired in future using the acceleration multiplier attribute from #9293). Heat dissipations were divided by the same amount, so ship heat maxima are unchanged.
The multipliers were derived from the ones in the previous PR by halving the distance from 1 (so, 1.2 -> 1.1, 1.8 -> 1.4, 2 -> 1.5, 3 -> 2, etc.). Some ships were slightly tweaked past this (+ or - .05 to the multiplier for a handful of ships), in cases where the multiplier felt too large or too small for the ship's design and/or role, and this has delibarately created some overlap between categories, in cases where a particular ship is especially large or small for its category.

Ships with integrated turn and thrust and no engines received x1.5 to mass, drag, turn and thrust and x(1/1.5) heat dissipation for consistency.

## Reasoning
It has been widely agreed that steeper scaling in handling and mass is something we want, so I'll mostly focus this section on why I think this set of changes is the best solution to that rather than on why that's something that's desired.

Progression of mobility with size should, in my view, mean within categories as well as between them. Approaches that simply apply a modifier based solely on category result in huge steps up between the boundaries of categories, and leave the issue of 'larger ships are always better' within categories. That's why, for example, the Protector's mobility is nerfed a lot less in this PR than the Carrier's.

The previous PR (#9451) extended the range as much as possible while keeping ships comfortable to pilot. This turned out to actually be too much of a buff to small ships and too much of a nerf to large ones, so this PR's smaller changes, in addition to being better for balance, have the nice side effect of very definitely keeping everything comfortable to fly.

Heavy Freighters received adjustments in line with medium warships rather than with heavy warships. This is because heavy freighters generally have similar outfit space numbers to medium warships, and because heavy freighters definitely do not need to be any slower than they already are.

## Testing Done
Two playthroughs of FW (one of FWR and one of FWC), plus extensive miscellaneous duelling in Omnis.

## Save File
Any, or better yet a new pilot (as old ships won't be updated)

## Todo:

- [ ] Test campaign combat balance

### CODE REVIEW

**Comment by Quantumshark** [comment]
The line count is different because last time I missed the Schist variants and the shard fabricator submunition.

**Review by Hurleveur — COMMENTED** [review]
loving the idea but yeah we probably want some ppl to test this to get a feel of it
right now the speed of things dont matter so much for the battle AI except for making them stack up a little bit more, maybe

File: data/hai/hai ships.txt [hunk 1]
```diff
@@ -709,10 +709,10 @@ ship "Pond Strider"
 		"hull" 3000
 		"required crew" 9
 		"bunks" 18
-		"mass" 168
-		"drag" 3.55
+		"mass" 429
```
[the remark this answers is not in the document]
Comment (Quantumshark) [reply data/hai/hai ships.txt:712]: It's still pretty light for its size, it's just that ships of that size are heavier now. 785.1 tons stock, compared to 1043.2 for the Ladybug or 1227.9 for the Sea Scorpion.

File: data/_deprecated/deprecated ships.txt [hunk 2]
```diff
@@ -212,9 +212,9 @@ ship "Korath World-Ship"
 		hull 34000
 		"required crew" 794
 		"bunks" 1492
-		"mass" 1735
-		"drag" 21
-		"heat dissipation" .4
+		"mass" 4440
+		"drag" 43
+		"heat dissipation" .2
```
[the remark this answers is not in the document]
Comment (Quantumshark) [reply data/_deprecated/deprecated ships.txt:217]: If we're keeping them around, we ought to keep them consistent with across-the-board changes to current content too, I'd say.

[the remark this answers is not in the document]
Comment (Hurleveur) [reply data/_deprecated/deprecated ships.txt:217]: I thought it was just for old plugins and such
which if they still use those the updates wouldnt have much sense for them

**Comment by Petersupes** [comment]
Overall these changes look good, the speeds of the different categories definitely feel correct. Handling I think has missed its mark in regards to the larger ships. Forward guns for heavy ships should be most effective attacking other heavy ships, and should be only effective against small ships like interceptors when they're attacking head on, but fairly useless when they're moving around the ship, leaving turrets, with their ability to turn faster than the ship their attached to, to be the main issue for smaller attackers. But in testing I found it very impractical for interceptors to reliably get around the forward guns of some HWs, in particular the stock cruiser and leviathan, both of which use ion engines and so should be quite easy to get past. A flatter scaling for steering could probably help with this.

File: data/_deprecated/deprecated ships.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Quantumshark) [reply data/_deprecated/deprecated ships.txt:217]: I'll wait and see what the devs think.

**Comment by Quantumshark** [comment]
I've tested this, and the Vanguard, which has the highest stock turn rate of any human HW, can still be beaten with a moderately well-built interceptor and good piloting, by avoiding its guns. If it's possible to do that against the fastest-turning heavy warship, it must presumably also be possible to do so against the others. Obviously you'll still be taking fire from missiles and turrets, but that's not anything to do with ship turn rate.

**Comment by Petersupes** [comment]
I definitely agree that HWs should win against smaller ships, just that they should do so using their bulk and their turrets. Upon further testing I've found the problem was less what I was testing against but what I was testing with. I wanted to test the stock ships to see if they were at least adequate at their job and the stock berserker is weirdly slow acceleration wise compared to other interceptors which all are capable of getting past the forward guns, even the fury which is supposed to have the worst maneuverability. This appears to just be a problem with the berserker's stock engines so nothing for this pr to worry about.

File: data/_deprecated/deprecated ships.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Quantumshark) [reply data/_deprecated/deprecated ships.txt:217]: Alternatively, I could rescale the masses of deprecated ships to preserve their prior handling, rather than updating them as in the PR? I don't think leaving them unchanged is a good idea, since because of the engine changes that'd make them 50% faster.

[the remark this answers is not in the document]
Comment (Hurleveur) [reply data/_deprecated/deprecated ships.txt:217]: thing is we're not going to change any ships acquired prior to the PR anyway
I mean change it if you want

[the remark this answers is not in the document]
Comment (Quantumshark) [reply data/_deprecated/deprecated ships.txt:217]: At this point we're getting into the question of what exactly `deprecated ships.txt` is for - it's not needed to avoid breaking already-acquired ships, it's only there for backwards compatibility with missions in plugins. I figure if it's worth keeping it around, it's also worth keeping it updated.

[the remark this answers is not in the document]
Comment (Zitchas) [reply data/_deprecated/deprecated ships.txt:217]: I was generally of the opinion that the point was to keep old ships as they were without any significant changes. If it's a change that might affect how a ship is outfitted or performs, then it shouldn't be happening to deprecated ships. Either of those risks making ships in plugins broken, and avoiding that is the entire point of deprecated ships.

The point of deprecated ships is to lock their stats at the point when the replaced ship diverges sufficiently from the original to risk stuff based on it breaking. Once forked, it should no longer be changed baring mechanical changes to the game that render it utterly non-functional. (ex. If we changed heat mechanics such that this ship would just take off and fry instantly and permanently, then that would merit a change; and even then I would say that it should be the minimum change required to allow it to perform the same as it did prior to the mechanical change, even if that differs significantly from how the parallel vanilla ship performs.

tldr: Deprecated ships are locking the ship to that point in time, and the only changes that should happen to it are ones that keep it functioning as close to exactly as it was at the point in time it was deprecated in the face of changes to the mechanics. Nothing else.

[the remark this answers is not in the document]
Comment (Quantumshark) [reply data/_deprecated/deprecated ships.txt:217]: That's a reasonable point. In which case, I'll update this accordingly, to preserve prior performance of deprecated ships as it would have been without the engine tweak.

**Comment by MasterOfGrey** [comment]
While I still believe that this game is going to have issues for so long as mass continues to be arbitrarily determined...

...this is probably the most coherent version of reworked arbitrary values that I've seen thus far.

So on that basis, this is better than the status quo and I support it.

**Review by MasterOfGrey — APPROVED** [review]

**Review by Hurleveur — APPROVED** [review]
I tested this out a bit, and definitely like the direction in which its going, small ships are easily more manoeuvrable and need to dedicate less space to thrusting in order to achieve it, and some HW are not overly agile anymore

**Review by mOctave — APPROVED** [review]
These changes aren't really noticeable unless you've been playing a lot with the current "balance." In fact, I think they make the movement in the game _more_ natural-looking. They also do a great job of helping light ships outperform heavier ones in speed.

**Review by Saugia — APPROVED** [review]
Did some testing over the last few weeks; played through some of the main campaign, tested ships individually and as a whole (I bought every ship and flew them around to test this as well).

The only noticeable stand-outs is that the ships on each end of the scale got either a bit faster or slower (i.e., the Combat Drone, and Large Korath ships), which is partially the intention here.

Many ships in-between have been shifted around slightly, but in a good way, which adds more variance in speeds between class (and more consistent comparisons therein). 

I would agree with m0ctave's comment - the changes, as a whole, are slight adjustments, comes across more natural and nothing is drastic enough where it breaks anything. I believe this is a good change; we'll see how things play out.

**Comment by 43897689431805741092** [comment]
Have the firing and hit forces of weapons been reviewed with respect to these changes? Can e.g. Hai Trackers still sufficiently disrupt the ships they were intended to before (assuming there were any particular intentions in the first place)? I haven't seen any mention of force attributes with respect to these changes, so it appears to have been overlooked.

Personally, I expect that, with the way ship masses grow more exponentially than before (I think?), things will behave better, not worse, with heavy ships actually having decent resistance to being pushed around that feels in line with what one would intuitively expect them to have.

Something that may be considered notable is that it should make human-space Korath raiders harder to loot by preventing them from jumping away using Particle Cannons or Trackers. I mean this still works, but requires a lot more of them, so disabling a Korath ship in the mid-game requires a bigger investment more proportional to the reward (i.e. a Jump Drive).

There was another stat I was thinking earlier, in addition to the force ones, but I can't remember now...

**Comment by Zitchas** [comment]
> Something that may be considered notable is that it should make human-space Korath raiders harder to loot by preventing them from jumping away using Particle Cannons or Trackers. I mean this still works, but requires a lot more of them, so disabling a Korath ship in the mid-game requires a bigger investment more proportional to the reward (i.e. a Jump Drive).

This is most definitely a good thing in my books. Taking down a Korath ship in the early and mid game should be very hard, very rare, and require deliberate and significant effort. For context, the Syndicate, which owns almost this entire region with immense resources and had direct access to basically camp out this area, had acquired <10 JD in all the years this raiding had been going on.

**Comment by Quantumshark** [comment]
> Have the firing and hit forces of weapons been reviewed with respect to these changes?

Yes, hit force and firing force were increased by 50% across the board, alongside thrust and turn. That +50%, which is also the mass increase which keeps performance invariant with the new engine changes, is in effect a new baseline, around which things which are not ship-specific were adjusted.

In the specific case of Korath ships, every Korath Exile ship other than the 'olofez is large enough to have a mass increase greater than +50%, and hence will be slower but more resistant to hit force than was previously the case.

**Comment by 43897689431805741092** [comment]
> There was another attribute I was thinking of earlier, in addition to the force ones, but I can't remember now...

Remembered this now: Has the scaling of optical tracking been reviewed?
That is, is it as ineffective versus 100 tons as it was before, and against larger ships does e.g. a Torpedo reach the point of 100% reliability against much smaller ships?
